Add documentation for the DynamicRowsRecord UI component (#5634)
magento/devdocs: guides/v2.2/ui_comp_guide/components/ui-dynamicrows-record.md

---
group: ui-components-guide
title: DynamicRowsRecord component
contributor_name: Atwix
contributor_link: https://www.atwix.com/
---

> The DynamicRowsRecord component is a container of record fields. The DynamicRowsRecord should be used as a child of [DynamicRows]({{ page.baseurl }}/ui_comp_guide/components/ui-dynamicrows.html) component.

Reviewer: ```suggestion
The DynamicRowsRecord component is a container of record fields. The DynamicRowsRecord should be used as a child of the [DynamicRows]({{ page.baseurl }}/ui_comp_guide/components/ui-dynamicrows.html) component.
```

## Configuration options

| Option | Description | Type | Default |
| --- | --- | --- | --- |

> | `disabled` | Initial component's state. When set to `true`, users can't take action on the element. | Boolean | `true` |

Reviewer: ```suggestion
| `disabled` | The initial state of the component. When set to `true`, users cannot take action on the element. | Boolean | `true` |
```

> | `headerLabel` | The label of the record. It is used as a record `label` in case if the `label` option is not initialized. | String | `''` |

Reviewer 2: Does that mean that `headerLabel` value is just a fallback for `label` value? Are you sure that's the only purpose of this field? 
Maybe it is, I'm just asking because having this flow looks a little bit weird.

Reviewer 2: Thank you for the clarification

> | `headerLabel` | The label of the record. It is used as a record `label` in case if the `label` option is not initialized. | String | `''` |

Reviewer: ```suggestion
| `headerLabel` | The label of the record. It is used as a record `label` when the `label` option is not initialized. | String | `''` |
```

| `label` | The label of the record. | String | `''` |
| `visible` | Initial component's visibility. When set to `false`, the `"display: none"` CSS style is added to the component's DOM block. | Boolean | `true` |

## Sources files

Extends [`UiCollection`]({{ page.baseurl }}/ui_comp_guide/concepts/ui_comp_uicollection_concept.html):

- [app/code/Magento/Ui/view/base/web/js/dynamic-rows/record.js]({{ site.mage2bloburl }}/{{page.guide_version}}/app/code/Magento/Ui/view/base/web/js/dynamic-rows/record.js)

## Examples

```xml

>     <form>

Reviewer: Adjust the indentation of the example so that the `<form>` element has no preceding spaces.
